server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/LinshareBlobExportMechanismIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+import org.apache.james.utils.JmapGuiceProbe;
+import org.apache.james.utils.WebAdminGuiceProbe;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.awaitility.core.ConditionFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.common.base.Strings;
+
+import io.restassured.RestAssured;
+import io.restassured.parsing.Parser;
+import io.restassured.specification.RequestSpecification;
+
+public abstract class LinshareBlobExportMechanismIntegrationTest {
+    private static final String LINSHARE_DOMAIN = "linshare.org";
+    private static final String USERNAME_1 = USER_1.getUsername();
+    private static final String PASSWORD_1 = USER_1.getPassword();
+    private static final String USERNAME_2 = USER_2.getUsername();
+    private static final String PASSWORD_2 = USER_2.getPassword();
+    private static final ConditionFactory WAIT_TWO_MINUTES = calmlyAwait.atMost(Duration.TWO_MINUTES);
+    private static final String SUBJECT = "This mail will be restored from the vault!!";
+    private static final String MATCH_ALL_QUERY = "{" +
+        "\"combinator\": \"and\"," +
+        "\"criteria\": []" +
+        "}";
+    private static final ExportRequest EXPORT_ALL_USER_1_MESSAGES_TO_USER_2 = ExportRequest
+        .userExportFrom(USERNAME_1)
+        .exportTo(USERNAME_2)
+        .query(MATCH_ALL_QUERY);
+
+    private AccessToken user1AccessToken;
+    private AccessToken user2AccessToken;
+    private GuiceJamesServer jmapServer;
+    private RequestSpecification webAdminApi;
+
+    @RegisterExtension
+    private IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+
+    @BeforeEach
+    void setup(GuiceJamesServer jmapServer) throws Throwable {
+        this.jmapServer = jmapServer;
+
+        jmapServer.getProbe(DataProbeImpl.class)
+            .fluent()
+            .addDomain(LINSHARE_DOMAIN);
+
+        RestAssured.requestSpecification = jmapRequestSpecBuilder
+            .setPort(jmapServer.getProbe(JmapGuiceProbe.class).getJmapPort())
+            .build();
+        RestAssured.defaultParser = Parser.JSON;
+
+        webAdminApi = WebAdminUtils.spec(jmapServer.getProbe(WebAdminGuiceProbe.class).getWebAdminPort());
+
+        MailboxProbe mailboxProbe = jmapServer.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox("#private", USERNAME_1, DefaultMailboxes.INBOX);
+
+        user1AccessToken = authenticateJamesUser(baseUri(jmapServer), USERNAME_1, PASSWORD_1);
+        user2AccessToken = authenticateJamesUser(baseUri(jmapServer), USERNAME_2, PASSWORD_2);
+    }
+
+    @Test
+    void exportShouldShareTheDocumentViaLinShareWhenJmapDelete() throws Exception {
+        user2SendMessageToUser1();
+
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(user1AccessToken).size() == 1);
```

Comment:
2 minutes!? does it need to be that long?

Reply:
Probably not !